api/src/main/java/org/pmiops/workbench/api/CohortsController.java

```diff
@@ -186,12 +197,51 @@ public Cohort apply(org.pmiops.workbench.db.model.Cohort cohort) {
     // 3. Otherwise, query BigQuery for participant IDs using SQL constructed from the cohort
     // criteria, subject to pagination; remove IDs for participants that had a cohort status not
     // included in the status filter.
-    return ResponseEntity.ok(new MaterializeCohortResponse());
+
+    Workspace workspace = workspaceService.getRequired(workspaceNamespace, workspaceId);
+    CdrVersion cdrVersion = workspace.getCdrVersion();
+    if (request.getCdrVersionName() != null) {
+      cdrVersion = cdrVersionDao.findByName(request.getCdrVersionName());
```

Comment:
Is it possible to request a cdrVersionName that doesn't belong to the workspaceNamespace and workspaceId? Should we check that the workspace.getCdrVersion() and cdrVersionDao.findByName(request.getCdrVersionName()) match?

Reply:
Yes, it's possible.

You can imagine a scenario where you create a cohort with CDR version 1, then when CDR version 2 comes along, you change your workspace to CDR version 2... but you still want to be able to analyze the data from CDR version 1 sometimes (perhaps comparing it to CDR version 2.)

So I don't think we want a check here.